addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/service/SmappeeService.java

```diff
@@ -0,0 +1,294 @@
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.apache.commons.httpclient.params.HttpMethodParams;
+import org.apache.commons.io.IOUtils;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SamppeeService} handles the connection to the Smappee API
+ *
+ * @author Niko Tanghe
+ */
+
+public class SmappeeService {
+
+    private static Logger logger = LoggerFactory.getLogger(SmappeeService.class);
+
+    private long refreshInterval;
+    private int retry;
+
+    private String client_id;
+    private String client_secret;
+    private String username;
+    private String password;
+    private String serviceLocationName;
+
+    private String serviceLocationId;
+
+    private boolean isInitialized;
+
+    private final Gson gson;
+
+    private String accessToken;
+    private DateTime accessTokenValidity;
+    private String refreshToken;
+
+    public SmappeeService(String client_id, String client_secret, String username, String password,
+            String serviceLocationName, long refreshInterval) {
+        this.serviceLocationName = serviceLocationName;
+        this.client_id = client_id;
+        this.client_secret = client_secret;
+        this.username = username;
+        this.password = password;
+
+        this.retry = 1;
+        this.refreshInterval = refreshInterval;
+
+        this.gson = new Gson();
+
+        this.isInitialized = false;
+    }
+
+    public void startAutomaticRefresh(ScheduledExecutorService scheduledExecutorService,
+            final ReadingsUpdate readingsUpdate) {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    SmappeeDeviceReading readings = getDeviceReadings();
+                    if (readings.consumptions.length > 0) {
+                        readingsUpdate.newState(readings);
+                    }
+                } catch (InvalidConfigurationException e) {
+                    readingsUpdate.invalidConfig();
+                }
+            }
+        };
+
+        scheduledJob = scheduledExecutorService.scheduleAtFixedRate(runnable, 0, this.refreshInterval,
+                TimeUnit.MILLISECONDS);
+    }
+
+    ScheduledFuture<?> scheduledJob;
+
+    public void stopAutomaticRefresh() {
+        scheduledJob.cancel(true);
+    }
+
+    public SmappeeDeviceReading getDeviceReadings() throws InvalidConfigurationException {
+        int currentTry = 0;
+        do {
+            try {
+                DateTime nowUtc = DateTime.now(DateTimeZone.UTC).minusMinutes(1);
+                DateTime nowUtcMinus20Min = DateTime.now(DateTimeZone.UTC).minusMinutes(20);
+
+                String nowUtcMillis = String.valueOf(nowUtc.getMillis());
```

Comment:
You can directly use the `nowUtc.getMillis()` the string concatenation it will do the to string automatically, making the code cleaner here.

Reply:
it's a different type ? (long -> String)

Reply:
I mean you can write: `String responseReadings = GetData("/dev/v1/servicelocation/" + this.serviceLocationId + "/consumption?aggregation=1&from=" + nowUtcMinus20Min.getMillis() + "&to=" + nowUtc.getMillis());`